sdk/src/main/java/io/opentdf/platform/sdk/TDF.java

```diff
@@ -164,57 +171,111 @@ PolicyObject createPolicyObject(List<String> attributes) {
         }
 
         private static final Base64.Encoder encoder = Base64.getEncoder();
-        private void prepareManifest(Config.TDFConfig tdfConfig) {
+        private void prepareManifest(Config.TDFConfig tdfConfig, SDK.KAS kas) {
             manifest.encryptionInformation.keyAccessType = kSplitKeyType;
             manifest.encryptionInformation.keyAccessObj =  new ArrayList<>();
 
             PolicyObject policyObject = createPolicyObject(tdfConfig.attributes);
             String base64PolicyObject  = encoder.encodeToString(gson.toJson(policyObject).getBytes(StandardCharsets.UTF_8));
             List<byte[]> symKeys = new ArrayList<>();
+            Map<String, Config.KASInfo> latestKASInfo = new HashMap<>();
+            if (tdfConfig.splitPlan.isEmpty()) {
+                // Default split plan: Split keys across all KASes
+                List<Config.SplitStep> splitPlan = new ArrayList<>(tdfConfig.kasInfoList.size());
+                for (Config.KASInfo kasInfo : tdfConfig.kasInfoList) {
+                    Config.SplitStep step = new Config.SplitStep(kasInfo.URL, "");
+                    if (tdfConfig.kasInfoList.size() > 1) {
+                        step.splitID = String.format("s-%d", splitPlan.size());
+                    }
+                    splitPlan.add(step);
+                    if (kasInfo.PublicKey != null && !kasInfo.PublicKey.isEmpty()) {
+                        latestKASInfo.put(kasInfo.URL, kasInfo);
+                    }
+                }
```

Comment:
I think the other clients create a split, not shared key by default. you'll have to add a counter 

```suggestion
                int i = 0;
                for (Config.KASInfo kasInfo : tdfConfig.kasInfoList) {
                    Config.SplitStep step = new Config.SplitStep(kasInfo.URL, "");
                    if (tdfConfig.kasInfoList.size() > 1) {
                        step.splitID = String.format("s-%d", i++);
                    }
                    splitPlan.add(step);
                    if (kasInfo.PublicKey != null && !kasInfo.PublicKey.isEmpty()) {
                        latestKASInfo.put(kasInfo.URL, kasInfo);
                    }
                }
```